safehttp/form.go

```diff
@@ -0,0 +1,253 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package safehttp
+
+import (
+	"fmt"
+	"mime/multipart"
+	"strconv"
+)
+
+// Form contains parsed data either from URL's query or from
+// form parameters, part of the body of POST, PATCH or PUT requests that are not
+// multipart requests. The fields are only available after parsing the form,
+// through getter functions that specify the type. If parsing failed, Form will
+// be set to nil. Field err will only be set if
+// an error occurs when the user tries to access a parameter.
+type Form struct {
+	values map[string][]string
+	err    error
+}
```

Comment:
Keep the definition of the struct next to the methods (i.e. move `MultipartForm` somewhere else/below).

Reply:
Resolved in [c5059f7](url)